lib/src/acter_avatar.dart

```diff
@@ -162,245 +126,242 @@ class _ActerAvatar extends State<ActerAvatar> {
     }
   }
 
-  String? secTooltipMsg() {
-    if (widget.avatarsInfo != null || widget.avatarsInfo!.isNotEmpty) {
-      switch (widget.secondaryToolTip) {
-        case TooltipStyle.DisplayName:
-          return widget.avatarsInfo![0].displayName ??
-              widget.avatarsInfo![0].uniqueId;
-        case TooltipStyle.UniqueId:
-          return widget.avatarsInfo![0].uniqueId;
-        case TooltipStyle.Combined:
-          var message = widget.avatarsInfo![0].uniqueName ??
-              widget.avatarsInfo![0].uniqueId;
-          if (widget.avatarsInfo![0].displayName != null) {
-            message =
-                '${widget.avatarsInfo![0].displayName} (${widget.avatarsInfo![0].uniqueId})';
-          }
-          return message;
-        case TooltipStyle.None:
-          return null;
-      }
-    } else {
-      return null;
+  Widget tooltipMsgWithChild(AvatarInfo avatar, Widget child) {
+    switch (avatar.tooltip) {
+      case TooltipStyle.DisplayName:
+        return Tooltip(
+          message: avatar.displayName ?? avatar.uniqueId,
+          child: child,
+        );
+      case TooltipStyle.UniqueId:
+        return Tooltip(message: avatar.uniqueId, child: child);
+      case TooltipStyle.Combined:
+        var message = avatar.uniqueName ?? avatar.uniqueId;
+        if (avatar.displayName != null) {
+          message = '${avatar.displayName} (${avatar.uniqueId})';
+        }
+        return Tooltip(message: message, child: child);
+      case TooltipStyle.None:
+        return child;
     }
   }
 
   Widget renderWithAvatar(BuildContext context, ImageProvider avatar) {
-    switch (widget.mode) {
-      case DisplayMode.DM:
-        return CircleAvatar(
-          foregroundImage: avatar,
-          onForegroundImageError: avatarError,
-          radius: widget.size ?? 24,
-        );
-      case DisplayMode.Space:
-        return Stack(
-          clipBehavior: Clip.none,
-          children: <Widget>[
-            Container(
-              height: widget.size ?? 48,
-              width: widget.size ?? 48,
-              decoration: BoxDecoration(
-                borderRadius: BorderRadius.circular(6.0),
-                image: DecorationImage(
-                  fit: BoxFit.cover,
-                  image: avatar,
-                  onError: avatarError,
+    return widget.options.when(
```

Comment:
I couldn't find any valid and easier approach that vehemently checks dart object initialisation by which constructor type. ```freezed``` seems to provide all necessary boilerplate generated methods, essentially making it easier to check object initialisation by constructor type. Although I believe this can be simplified further by wrapping into sub-widget methods as per your feedback.